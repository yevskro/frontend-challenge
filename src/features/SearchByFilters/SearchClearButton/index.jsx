import React from 'react';
import styled from 'styled-components';

function SearchClearButton({ onClick }) {
  return <StyledSearchButton onClick={onClick}>Clear</StyledSearchButton>;
}

const StyledSearchButton = styled.button`
  text-align: center;
  font-size: ${({ theme }) => theme.fontSize.medium};
  color: ${({ theme }) => theme.color.neutral.darkGrayishCyan};

  &:hover {
    color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
    text-decoration: underline;
    cursor: pointer;
  }
`;

export default SearchClearButton;
