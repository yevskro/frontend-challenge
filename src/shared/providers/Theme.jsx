import React from 'react';
import { ThemeProvider } from 'styled-components';
/* theme developed based on the style-guide */
const theme = {
  color: {
    primary: {
      desaturtedDarkCyan: 'hsl(180, 29%, 50%)',
    },
    neutral: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilterTablets: 'hsl(180, 31%, 95%)',
      darkGrayishCyan: 'hsl(180, 8%, 52%)',
      veryDarkGrayishCyan: 'hsl(180, 14%, 20%)',
    },
    shadow: 'rgba(94, 140, 160, 0.15)',
  },
  fontSize: {
    xxsmall: '0.7rem',
    xsmall: '0.8rem',
    small: '0.9rem',
    medium: '1rem',
    large: '1.2rem',
  },
  fontWeight: {
    thin: '400',
    normal: '500',
    bold: '600',
  },
  fontFamily: '"Spartan", "sans-serif"',
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
