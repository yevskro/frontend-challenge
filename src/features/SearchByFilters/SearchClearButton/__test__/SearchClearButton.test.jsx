import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Theme from '../../../../shared/providers/Theme';
import SearchClearButton from '..';

test('renders', () => {
  render(
    <Theme>
      <SearchClearButton />
    </Theme>
  );
  expect(screen.getByText('Clear')).toBeInTheDocument();
});

test('onClick callback invoked', () => {
  const mock = jest.fn();
  render(
    <Theme>
      <SearchClearButton onClick={mock} />
    </Theme>
  );

  userEvent.click(screen.getByText('Clear'));
  expect(mock).toHaveBeenCalledTimes(1);
});

test('snapshot', async () => {
  const mock = jest.fn();
  const tree = renderer
    .create(
      <Theme>
        <SearchClearButton onClick={mock} />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
