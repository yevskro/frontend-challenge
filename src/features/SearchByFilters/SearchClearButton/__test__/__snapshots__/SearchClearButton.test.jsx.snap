// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  text-align: center;
  font-size: 1rem;
  color: hsl(180,8%,52%);
}

.c0:hover {
  color: hsl(180,29%,50%);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  Clear
</button>
`;
